### PR TITLE
Update the documentation to the new daemon format

### PR DESCRIPTION
**Pull request by haixuanTao**
Update the documentation to the new API and the new event stream.

I have updated the python example as it appears in the documentation as link.

### CODE REVIEW

File: docs/src/c-api.md [hunk 1]
```diff
@@ -70,9 +35,9 @@ The custom node API allow you to integrate `dora` into your application. It allo
 void *dora_context = init_dora_context_from_env();
 ```
 
-#### `dora_next_input`
+#### `dora_next_event`
 
-`dora_next_input` waits for the next input. To extract the input ID and data, use `read_dora_input_id`  and `read_dora_input_data` on the returned pointer.
+`dora_next_event` waits for the next input. To extract the input ID and data, use `read_dora_input_id`  and `read_dora_input_data` on the returned pointer.
```
Comment (phil-opp) [inline docs/src/c-api.md:40]:
```suggestion
`dora_next_event` waits for the next event (e.g. an input). Use `read_dora_event_type` to read the event's type. Inputs are of type `DoraEventType_Input`. To extract the ID and data of an input event, use `read_dora_input_id`  and `read_dora_input_data` on the returned pointer. It is safe to ignore any events and handle only the events that are relevant to the node.
```

File: docs/src/rust-api.md [hunk 2]
```diff
@@ -4,25 +4,10 @@
 
 The operator API is a framework for you to implement. The implemented operator will be managed by `dora`. This framework enable us to make optimisation and provide advanced features. It is the recommended way of using `dora`.
 
-An operator requires to be registered and implement the `DoraOperator` trait. It is composed of an `on_input` method that defines the behaviour of the operator when there is an input.
+An operator requires to be registered and implement the `DoraOperator` trait. It is composed of an `on_event` method that defines the behaviour of the operator when there is an event such as receiving an input for exemple.
```
Comment (phil-opp) [inline docs/src/rust-api.md:7]: Typo:

```suggestion
An operator requires to be registered and implement the `DoraOperator` trait. It is composed of an `on_event` method that defines the behaviour of the operator when there is an event such as receiving an input for example.
```

**Review by phil-opp — APPROVED** [review]

**Comment by phil-opp** [comment]
Looks good! Feel free to merge this.

**Comment by haixuanTao** [comment]
Just going to double check the documentation on a fresh install and merge it afer it.

**Pull request merged**